hazelcast/src/main/java/com/hazelcast/jet/impl/connector/StreamEventJournalP.java

```diff
@@ -369,7 +375,11 @@ private CompletableFuture<? extends ReadResultSet<? extends T>> readFromJournal(
                 @Nonnull JournalInitialPosition initialPos,
                 @Nonnull EventTimePolicy<? super T> eventTimePolicy,
                 @Nonnull SupplierEx<Permission> permissionFn
-                ) {
+        ) {
+            if (dataLinkName != null && clientXml != null) {
```

Comment:
You can create a function for checking dataLinkName and clientXxml

boolean isBothSet(datalinkName, clientXML) {
 return idataLinkName != null && clientXml != null)
}
boolean isAnySet(datalinkName, clientXML) {
 return idataLinkName != null || clientXml != null)
}

Reply:
58878a42e14